From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Walker <andrew_walker@jabil.com>
Date: Thu, 17 Sep 2026 14:28:00 -0400
Subject: [PATCH] Jetson/board.py: search /usr/lib/tegra for DTBs

JBLR mounts the ESP vfat partition on /boot, hiding rootfs /boot/dtb.
l4t-launcher-extlinux-dtb-extra mirrors device trees to /usr/lib/tegra.

Upstream-Status: Pending
Signed-off-by: Andrew Walker <andrew_walker@jabil.com>
---
 opt/nvidia/jetson-io/Jetson/board.py | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/opt/nvidia/jetson-io/Jetson/board.py b/opt/nvidia/jetson-io/Jetson/board.py
index a1b2c3d..b2c3d4e 100644
--- a/opt/nvidia/jetson-io/Jetson/board.py
+++ b/opt/nvidia/jetson-io/Jetson/board.py
@@ -249,11 +249,19 @@ class Board(object):
         # Import platform specific data
         self.compat = dt.read_prop('compatible')
         self.model = dt.read_prop('model')
-        dtb_search_paths = [dtbdir, os.path.join(self.bootdir, 'devicetree')]
+        dtb_search_paths = [
+            dtbdir,
+            os.path.join(self.bootdir, 'devicetree'),
+            '/usr/lib/tegra/dtb',
+        ]
         if self.appdir:
             dtb_search_paths.append(os.path.join(self.appdir, 'boot/devicetree'))
         self.dtb = _board_get_dtb(self.compat, self.model, dtb_search_paths)
-        dtbo_search_paths = [self.bootdir, os.path.join(self.bootdir, 'devicetree')]
+        dtbo_search_paths = [
+            self.bootdir,
+            os.path.join(self.bootdir, 'devicetree'),
+            '/usr/lib/tegra/dtbo',
+        ]
         if self.appdir:
             dtbo_search_paths.append(os.path.join(self.appdir, 'boot'))
         dtbos = dtc.find_compatible_dtbo_files(self.compat.split(),
-- 
2.43.0
